pkg/controller/pxc/controller.go

```diff
@@ -78,12 +83,52 @@ func newReconciler(mgr manager.Manager) (reconcile.Reconciler, error) {
 
 // add adds a new Controller to mgr with r as the reconcile.Reconciler
 func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	if err := setupFieldIndexers(mgr); err != nil {
+		return errors.Wrap(err, "setup field indexers")
+	}
 	return builder.ControllerManagedBy(mgr).
 		Named(naming.OperatorController).
 		Watches(&api.PerconaXtraDBCluster{}, &handler.EnqueueRequestForObject{}).
+		Watches(&corev1.Secret{}, enqueuePXCReferencingSecret(mgr.GetClient())).
 		Complete(r)
 }
 
+func setupFieldIndexers(mgr manager.Manager) error {
```

Comment:
This function's purpose is mainly for setting up the secret field indexer, maybe we can reflect that to the name e.g. `setupSecretNameFieldIndexer`

Reply:
updated